--- lparser.c
+++ lparser.c
@@ -4933,14 +4933,15 @@
   /* Check if it is a classic goto label or dynamic goto */
   int is_dynamic = 0;
   if (ls->t.token != TK_NAME && ls->t.token != TK_CONTINUE) {
       is_dynamic = 1;
   } else if (ls->t.token == TK_NAME) {
       expdesc dummy;
       int varkind = searchvar(fs, ls->t.seminfo.ts, &dummy);
       int lookahead = luaX_lookahead(ls);
-      if (varkind >= 0 || lookahead == '.' || lookahead == '[' || lookahead == '(' || lookahead == '{' || lookahead == ':') {
+      if (varkind >= 0 || lookahead == '.' || lookahead == '[' || lookahead == '(' || lookahead == '{' || lookahead == ':' || lookahead == TK_DBCOLON) {
           is_dynamic = 1;
       }
   }
